core/src/main/scala/kafka/log/LogManager.scala

```diff
@@ -203,6 +208,11 @@ class LogManager(val logDirs: Array[File],
                          delay = InitialTaskDelayMs,
                          period = flushCheckpointMs,
                          TimeUnit.MILLISECONDS)
+      scheduler.schedule("kafka-delete-logs",
```

Comment:
The `deleteLogs` function swallows all exceptions and reports them. On IO exceptions on a particular log, it will retry later. The current implementation has a queue of logs already. It's semantically equivalent to `scheudler.schedule` afaik. I did not feel compelled to change.